src/main/java/org/elasticsearch/common/lucene/all/AllTermQuery.java

```diff
@@ -77,31 +81,43 @@ public AllTermSpanScorer scorer(LeafReaderContext context, Bits acceptDocs) thro
             protected float payloadScore;
             protected int payloadsSeen;
 
-            public AllTermSpanScorer(TermSpans spans, Weight weight, Similarity.SimScorer docScorer) throws IOException {
+            public AllTermSpanScorer(TermSpans spans, SpanWeight weight, Similarity.SimScorer docScorer) throws IOException {
                 super(spans, weight, docScorer);
                 positions = spans.getPostings();
             }
 
             @Override
-            protected boolean setFreqCurrentDoc() throws IOException {
-                if (!more) {
-                    return false;
-                }
-                doc = spans.doc();
+            protected void setFreqCurrentDoc() throws IOException {
                 freq = 0.0f;
                 numMatches = 0;
                 payloadScore = 0;
                 payloadsSeen = 0;
-                do {
-                    int matchLength = spans.end() - spans.start();
 
-                    freq += docScorer.computeSlopFactor(matchLength);
+                assert spans.startPosition() == -1 : "incorrect initial start position, spans="+spans;
```

Comment:
And i just did not have the time to yet yesterday remove the stupid asserts from SpanScorer. Please, lets not drag this stuff in again. If oyu want to push fine, but you will see a second push from me removing all this crap.